src/Avalonia.Controls/Platform/ITopLevelImpl.cs

```diff
@@ -22,6 +22,11 @@ public interface ITopLevelImpl : IDisposable
         /// </summary>
         Size ClientSize { get; }
 
+        /// <summary>
+        /// Gets the total size of the toplevel, excluding shadows.
+        /// </summary>
+        Size TotalSize { get; }
```

Comment:
Should this be a `Size?` due to the non-client size not being available on all platforms (i.e. Wayland if I understand correctly)?

Reply:
Not sure I understand what you mean. IIRC wayland forces windows to draw their own decorations so supporting this should be easy?

Reply:
Should this be a `Size?` ? 

i.e. nullable

Reply:
Ah yes, so Wayland shouldn't be a problem, but is it always possible to get the frame size on X11 with server-side decorations? If not, a nullable `Size` would be better.

Reply:
@ngarside wayland does not force client decoration. It forces the apps to support client decorations. Compositors that want server decorations announce that for apps. 

I understand that GNOME-folks tend to completely ignore anything but GNOME, but this behavior is not welcome here.

Reply:
Ah OK, didn't realise that was a gnome thing rather than wayland, changed to nullable

Reply:
:+1:

Reply:
Sorry if I sounded rude, but it's a touchy subject that causes SSD vs. CSD flame pretty much every time when wayland is mentioned. Since Avalonia is cross-platform first, we just want to support all desktop environments instead of preferring one and breaking UX for others.